hedera-node/docs/system-accounts-operations.md

```diff
@@ -0,0 +1,184 @@
+- 0.0.55 - address book admin;
+- 0.0.57 - exchange rates admin;
+- 0.0.58 - freeze admin;
+- 0.0.59 - system delete admin;
+- 0.0.60 - system undelete admin;
+- 0.0.800 - staking reward account;
+- 0.0.801 - node reward account.
+
+Addresses in range 1 → 9 are special addresses on which the Ethereum precompiled contracts exist. This is valid for both Ethereum and Hedera.
+
+Lastly, the system accounts can be divided in these two groups:
+- addresses in range 0 → 750 - accounts that reject hbar transfers. An exception is the HTS precompiled contract that can receive transfers.
+- addresses in range 751 → 1000 - accounts which may receive transfers.
+
+## Rationale
+As a result of the specifics of the system accounts in Hedera described in the section above there are some differences when calling operations on system accounts in Hedera and in Ethereum via smart contract executions.
+This doc covers these differences and describes the expected behavior.
+
+## Operations
+
+Legend of the symbols used in the tables below:
+- ✅ - the operation was successful;
+- ❌ - the operation failed;
+- ❗ - something specific to note when executing an operation.
+
+### ExtCode operations
+
+`ExtCodeSize`
+
+**Ethereum:** Get size of an account’s code. If there’s no code, it returns 0.
+
+**Hedera:** For addresses in range 0x0 → 0x3E8 (0.0.0 → 0.0.1000) it returns 0.
+
+`ExtCodeCopy`
+
+**Ethereum:** Copy an account’s code to memory. If a contract doesn’t exist on the address, it copies empty bytes.
+
+**Hedera:** For addresses in range 0x0 → 0x3E8 (0.0.0 → 0.0.1000) it copies empty bytes.
+
+`ExtCodeHash`
+
+**Ethereum:** Hash of the chosen account's code, the empty hash (0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470) if the account has no code, or 0 if the account does not exist or has been destroyed.
+
+**Hedera:** For addresses in range 0x0 → 0x3E8 (0.0.0 → 0.0.1000) it returns the empty hash.
+
+### Balance operation
+
+**Ethereum:** Returns balance or 0 if an account or contract doesn’t exist on the address. This is valid for any address.
+
+**Hedera:**
+1. For addresses in range 0x0 → 0x2EE (0.0.0 → 0.0.750): returns 0 as if account doesn’t exist.
+2. For addresses in range 0x2EF → 0x3E8 (0.0.751 → 0.0.1000): returns balance or 0 if an account or contract doesn’t exist on the address. This behavior is the same for addresses above 0.0.1000.
+
+| Address                            | in Ethereum                                                                   | in Hedera                                                                     |
+|------------------------------------|-------------------------------------------------------------------------------|-------------------------------------------------------------------------------|
+| 0x0                                | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns 0 as if account doesn’t exist                                       |
```

Comment:
I believe we always return 0x if below 0.0.751

Reply:
As far as I see we always return 0 for EVM addresses below 751. We have a test for that - `Evm46ValidationSuite#testBalanceOfForSystemAccounts`.

Reply:
Q: Why is this?
@stoqnkpL Did we miss this scenario?
For example there are some system account with balance below 751, why not return the valid balance?
This is returning incorrect data if not.

Reply:
When we designed this we wanted the EVM to behave as if this account does not exist and can't be created. We intentionally decided to return 0 as if the account doesn't exist.

Reply:
Ah yes, so far as it can't receive balance then this is valid, but if it can this is invalid.